app/models/reports/percentage.rb

```diff
@@ -0,0 +1,35 @@
+module Reports::Percentage
+  def percentage(numerator, denominator, with_rounding: true)
+    return 0 if numerator.nil? || denominator.nil? || denominator == 0 || numerator == 0
+    if with_rounding
+      ((numerator.to_f / denominator) * 100).round(Reports::PERCENTAGE_PRECISION)
+    else
+      ((numerator.to_f / denominator) * 100)
+    end
+  end
+
+  def rounded_percentages(counts_hash)
```

Comment:
I like having a function to handle this, but I'm not sure it should be baked into the repository.  My concern is that it could lead to the repo returning the rounded values that have lost precision, when I think its more the responsibility of a higher layer to handle that.

In other words, I think the repo should be trusted to return the most precise number available, and then callers could hand it off to the rounded_percentage helper if they want to make sure graphs appear correct.

Reply:
We went back and forth between a helper and keeping it in the repository. We kept it inside the repository because:
- The other methods are already exposing rounded percentages, this just fixes a rounding issue these other methods also [potentially have](https://github.com/simpledotorg/simple-server/pull/3301)
- Leaving the rounding upto the caller leaves room for them to implement their own rounding procedure which might lead to the same problem we're seeing with the proliferation of different `percentage` methods

The one reason we can think of to not do this is if in the future someone needs raw percentages from the repo. Not sure if we should build for that use case yet.

-- priyanga and prabhanshu